middlewares/accesslog/logger.go

```diff
@@ -19,15 +25,36 @@ const (
 )
 
 // LogHandler will write each request and its response to the access log.
-// It gets some information from the logInfoResponseWriter set up by previous middleware.
-// Note: Current implementation collects log data but does not have the facility to
-// write anywhere.
 type LogHandler struct {
+	logger *logrus.Logger
+	file   *os.File
 }
 
 // NewLogHandler creates a new LogHandler
-func NewLogHandler() *LogHandler {
-	return &LogHandler{}
+func NewLogHandler(filePath string) (*LogHandler, error) {
+	if len(filePath) == 0 {
+		return nil, errors.New("Empty file name specified for accessLogsFile")
+	}
+
+	dir := filepath.Dir(filePath)
+
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		log.Errorf("Failed to create log path %s: %s", dir, err)
```

Comment:
Logging an error and returning an error normally do not go together; instead, we should return an error containing sufficient context and let the caller log.

Could you please pass the (de-capitalized) log message content to an `fmt.Errorf` call and just return that?

Same for lines 48:49.